install/migrations/update_10.0.x_to_11.0.0/databaseinstance.php

```diff
@@ -40,3 +40,20 @@
 
 // The size field seems to be unsused and rather a copy-paste from the glpi_databases table
 $migration->dropField('glpi_databaseinstances', 'size');
+
+// Block update if any database instance is currently unlinked
+$unlinked_it = $DB->request([
+    'SELECT' => ['id'],
+    'FROM'   => 'glpi_databaseinstances',
+    'WHERE' => [
+        'OR' => [
+            'itemtype' => '',
+            'items_id' => 0
+        ]
+    ]
+]);
+
+if (count($unlinked_it)) {
+    $unlinked_id_string = implode(', ', array_column(iterator_to_array($unlinked_it), 'id'));
+    throw new \RuntimeException("Some database instances are not linked to an item: {$unlinked_id_string}");
+}
```

Comment:
Having an orphan database instance can cause issues, but it should probably be handled the same way as we have to handle a database instance that is linked to an itemtype from a plugin that is no more available/activated. So, IMHO, blocking migrations is not necessary.
Also, blocking migration would really be a pain to handle for administrators, as they will probably not be able to know what they should do with orphan databases. Indeed, most of the time, only the user that created the data would really know what to do with this.

Reply:
If DatabaseInstance is supposed to be CommonDBChild (not even sure it is technically possible), it will be required to be linked.

I'm not sure it makes sense for DatabaseInstance to have its own entity column either. It should require a linked asset and use that entity/visibility restrictions.

Reply:
I agree it should not be orphan, but it is not up to the administrator to fix this to unlock the migration. Think about the administrators of the GLPI Network Cloud solution, they have no idea of what to do with this data, and they will probably not want to wait for every customer to fix its own data to be able to run the migration.

We already have similar cases of orphaned CommonDBChild instances when a plugin is disabled and its data not cleaned, we either ignore them, or trigger a warning. I consider that we should handle this case the same way.

Reply:
Agree, it should not blocking.
If preferred, we can keep an warning message for administrators. (via Migration::log)

Reply:
This isn't really the same as having a missing plugin. If a database instance was linked to a plugin item and the plugin is missing, that database is still linked. Doesn't matter that it is a broken link at runtime, it is still a link.

Having null/0 as the itemtype/items_id means there is no link at all and there should always be a link for database instances even if it isn't currently valid.

Reply:
The main issue here is that it is not acceptable to block migrations.

Reply:
We can't leave invalid data either that is going to cause issues later.

I'll close this and I won't work on the mess of allowing database instances to be transferred through the entity transfer feature.